ibtk/src/math/HierarchyMathOps.cpp

```diff
@@ -2053,20 +2096,40 @@ HierarchyMathOps::interp(const int dst_idx,
 
             d_patch_math_ops.interp(dst_data, src_data, patch);
 
-            if ((ln > d_coarsest_ln) && src_cf_bdry_synch)
+            if (ln > d_coarsest_ln)
             {
-                Pointer<OutersideData<NDIM, double> > os_data = patch->getPatchData(d_os_idx);
-                os_data->copy(*src_data);
+                if (src_cf_bdry_synch)
+                {
+                    Pointer<OutersideData<NDIM, double> > os_data = patch->getPatchData(d_os_idx);
+                    os_data->copy(*src_data);
+                }
+                if (dst_cf_bdry_synch)
+                {
+                    Pointer<OuternodeData<NDIM, double> > on_data = patch->getPatchData(on_idx);
+                    on_data->copy(*dst_data);
+                }
             }
         }
 
         // Synchronize the coarse-fine interface and deallocate temporary data.
-        if ((ln > d_coarsest_ln) && src_cf_bdry_synch)
+        if (ln > d_coarsest_ln && src_cf_bdry_synch)
         {
             xeqScheduleOutersideRestriction(src_idx, d_os_idx, ln - 1);
             level->deallocatePatchData(d_os_idx);
         }
+        if (ln + 1 <= d_finest_ln && dst_cf_bdry_synch)
+        {
+            xeqScheduleOuternodeRestriction(dst_idx, on_idx, ln);
+            d_hierarchy->getPatchLevel(ln + 1)->deallocatePatchData(on_idx);
+        }
+    }
+
+    // fixup data at hanging nodes.
+    if (dst_cf_bdry_synch)
```

Comment:
```suggestion
    // If the fine boundary represents the variable value at the coarse fine interface, then
    // do not overwrite values along the "fine side" of the coarse-fine interface.
    if (dst_cf_bdry_synch && !dst_var->fineBoundaryRepresentsVariable())
```
If the fine boundary represents the variable, we don't want to overwrite the fine